cmd/av/stack.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"emperror.dev/errors"
+	"github.com/spf13/cobra"
+	"strconv"
+)
+
+var stackCmd = &cobra.Command{
+	Use:   "stack",
+	Short: "managed stacked pull requests",
+}
+
+var stackBranchFlags struct {
+	// The parent branch to base the new branch off.
+	// By default, this is the current branch.
+	Parent string
+}
+var stackBranchCmd = &cobra.Command{
+	Use:   "branch <branch name>",
+	Short: "create a new stacked branch",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New("branch name is required")
+		}
+		return errors.New("unimplemented")
+	},
+}
+
+var stackSyncFlags struct {
+	// Set the parent of the current branch to this branch.
+	// This effectively re-roots the stack on a new parent.
+	Parent string
+	// If set, incorporate changes from the trunk (repo base branch) into the stack.
+	// Only valid if synchronizing the root of a stack.
+	// This effectively re-roots the stack on the latest commit from the trunk.
+	Trunk bool
+	// If set, do not push to GitHub.
+	NoPush bool
+	// If set, we're continuing a previous sync.
+	// TODO:
+	// 	 we might not actually need this, we can probably detect that
+	//   a sync needs to be completed automagically and do the right thing.
+	Continue bool
+}
+var stackSyncCmd = &cobra.Command{
+	Use:   "sync",
+	Short: "synchronize all stacked branches after the current branch",
```

Comment:
after or before? So if we do:
X -> Y -> Z
and we are on Y. Do we update Y with X or do we update Z from Y?

Reply:
I would imagine that you go a specific branch and sync everything up to this branch?

I guess the other alternative is that we push a commit to this branch and push it upwards (to the next PRs). 🤷

Reply:
It updates every branch *after* the current branch. I think that makes sense considering that you'll generally commit to a branch then update all the children.

I suspect you're thinking of a workflow where you edit `X`, then want to edit `Y`, then want to edit `Z`. With this it would be kind of difficult (you'd have to edit X, then run sync -- which rebases everything downstack, then checkout and edit Y, run sync, etc., which means you might be resolving a lot of unnecessary conflicts).

Yeah, let me think through this a bit more.

Reply:
I think we already discussed this offline.